src/pretix/base/datasync/datasync.py

```diff
@@ -0,0 +1,425 @@
+            mapped_objects: dict,
+            **kwargs,
+    ):
+        raise NotImplementedError()
+
+    def sync_object(
+            self,
+            inputs: dict,
+            mapping,
+            mapped_objects: dict,
+    ):
+        logger.debug("Syncing object %r, %r, %r", inputs, mapping, mapped_objects)
+        properties = self.get_properties(inputs, mapping.property_mapping)
+        logger.debug("Properties: %r", properties)
+
+        pk_value = self.get_field_value(inputs, {"pretix_field": mapping.pretix_pk})
+        if not pk_value:
+            return None
+
+        info = self.sync_object_with_properties(
+            pk_field=mapping.external_pk,
+            pk_value=pk_value,
+            properties=properties,
+            inputs=inputs,
+            mapping=mapping,
+            mapped_objects=mapped_objects,
+        )
+        OrderSyncLink.objects.create(
+            order=inputs.get(ORDER), order_position=inputs.get(ORDER_POSITION), sync_provider=self.identifier,
+            external_object_type=info.get('object_type'),
+            external_pk_name=info.get('pk_field'),
+            external_pk_value=info.get('pk_value'),
+            external_link_href=info.get('external_link_href'),
+            external_link_display_name=info.get('external_link_display_name'),
+        )
+        return info
+
+    def sync_order(self, order):
+        if not self.order_valid_for_sync(order):
+            logger.debug("Skipping order %r (not valid for sync)", order)
+            return
+
+        logger.debug("Syncing order %r", order)
+        positions = list(
+            order.all_positions.filter(item__admission=True)
+            .prefetch_related("answers", "answers__question")
+            .select_related(
+                "voucher",
+            )
+        )
+        order.synced_objects.filter(sync_provider=self.identifier).delete()
+        order_inputs = {ORDER: order, EVENT: self.event}
+        mapped_objects = {}
+        for mapping in self.mappings:
+            if mapping.pretix_model == 'Order':
+                mapped_objects[mapping.pk] = [
+                    self.sync_object(order_inputs, mapping, mapped_objects)
+                ]
+            elif mapping.pretix_model == 'OrderPosition':
+                mapped_objects[mapping.pk] = [
+                    self.sync_object({
+                        **order_inputs, EVENT_OR_SUBEVENT: op.subevent or self.event, ORDER_POSITION: op
+                    }, mapping, mapped_objects)
+                    for op in positions
+                ]
+            else:
+                raise SyncConfigError("Invalid pretix model '{}'".format(mapping.pretix_model))
+        self.finalize_sync_order(order)
+        return mapped_objects
+
+    """
+    Called after sync_object has been called successfully for all objects of a specific order. Can be used for saving
+    bulk information per order.
+    """
+    def finalize_sync_order(self, order):
+        pass
```

Comment:
Do you have a use case example in mind for this? With only order as input, I wonder how much useful things this function could possibly do